### PR TITLE
Add support for oodle compression

### PR DESCRIPTION
**Pull request by brunoanc**
Adds support for oodle compression. Tested on: https://github.com/PowerBall253/idSaveDecompressorLinux

### CODE REVIEW

File: linoodle.cpp [hunk 1]
```diff
@@ -16,14 +18,24 @@ class OodleWrapper {
...
     tDecompressFunc m_decompressFunc;
+    tCompressFunc m_compressFunc;
 };
 
 OodleWrapper g_oodleWrapper;
 extern "C" size_t OodleLZ_Decompress(uint8_t * srcBuf, size_t srcLen, uint8_t * dstBuf, size_t dstLen, int64_t unk1, int64_t unk2, int64_t unk3, int64_t unk4, int64_t unk5, int64_t unk6, int64_t unk7, int64_t unk8, int64_t unk9, int64_t unk10)
 {
     return g_oodleWrapper.Decompress(srcBuf, srcLen, dstBuf, dstLen, unk1, unk2, unk3, unk4, unk5, unk6, unk7, unk8, unk9, unk10);
 }
+extern "C" size_t OodleLZ_Compress(uint8_t * srcBuf, size_t srcLen, uint8_t * dstBuf, size_t dstLen, int64_t unk1, int64_t unk2, int64_t unk3, int64_t unk4, int64_t unk5, int64_t unk6, int64_t unk7, int64_t unk8, int64_t unk9, int64_t unk10)
```
Comment (McSimp) [inline linoodle.cpp]: From what I can tell from other sources on the internet (https://github.com/Crauzer/OodleSharp/blob/master/OodleSharp/Oodle.cs#L10, https://en.m.uesp.net/wiki/Online_Mod:DAT_File_Format, https://github.com/powzix/ooz/blob/master/kraken.cpp#L4288), this doesn't seem like the right signature. Are you sure that it works correctly?

Comment (brunoanc) [reply linoodle.cpp]: Huh, that's weird, it did work for me. On the DOOM Eternal projects the following are used:
```
typedef int WINAPI OodLZ_DecompressFunc(uint8* src_buf, int src_len, uint8* dst, size_t dst_size,
    int fuzz, int crc, int verbose,
    uint8* dst_base, size_t e, void* cb, void* cb_ctx, void* scratch, size_t scratch_size, int threadPhase);
```
```
typedef int WINAPI OodLZ_CompressFunc(
    int codec, uint8* src_buf, size_t src_len, uint8* dst_buf, int level,
    void* opts, size_t offs, size_t unused, void* scratch, size_t scratch_size);
```

I can change it to what's on the "official" documents, but it's weird that it worked.

**Comment by brunoanc** [comment]
I have changed the function to match the documentation provided, tested and it seems to work fine. Waiting for your review.

**Comment by brunoanc** [comment]
Also working on https://github.com/PowerBall253/EternalModLoaderCpp.

**Pull request merged**